### ISSUE TITLE
Emojis in README.rst makes the build fail with a UnicodeDecodeError

### ISSUE THREAD

**Issue by lrq3000** [pypa/build#599]
When a readme.rst includes emoji characters such as ⏳🛡🔮🚀, the build fails with the following error rtace:

```
* Creating virtualenv isolated environment...
* Installing packages in isolated environment... (cython>=3.0.0b2, setuptools>59, wheel>=0.37.0)
* Getting build dependencies for sdist...
Traceback (most recent call last):
  File "C:\Users\USERNAME\miniconda3\lib\site-packages\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
    main()
  File "C:\Users\USERNAME\miniconda3\lib\site-packages\pyproject_hooks\_in_process\_in_process.py", line 335, in main
    json_out['return_val'] = hook(**hook_input['kwargs'])
  File "C:\Users\USERNAME\miniconda3\lib\site-packages\pyproject_hooks\_in_process\_in_process.py", line 287, in get_requires_for_build_sdist
    return hook(config_settings)
  File "C:\Users\USERNAME\AppData\Local\Temp\build-env-4gyh6br5\lib\site-packages\setuptools\build_meta.py", line 341, in get_requires_for_build_sdist
    return self._get_build_requires(config_settings, requirements=[])
  File "C:\Users\USERNAME\AppData\Local\Temp\build-env-4gyh6br5\lib\site-packages\setuptools\build_meta.py", line 320, in _get_build_requires
    self.run_setup()
  File "C:\Users\USERNAME\AppData\Local\Temp\build-env-4gyh6br5\lib\site-packages\setuptools\build_meta.py", line 335, in run_setup
    exec(code, locals())
  File "<string>", line 72, in <module>
  File "C:\Users\USERNAME\miniconda3\lib\encodings\cp1252.py", line 23, in decode
    return codecs.charmap_decode(input,self.errors,decoding_table)[0]
UnicodeDecodeError: 'charmap' codec can't decode byte 0x8f in position 173: character maps to <undefined>

ERROR Backend subprocess exited when trying to invoke get_requires_for_build_sdist
```

Removing the emojis solves the issue.

Tested on build v0.10.0, Windows 10 x86.

**Comment by henryiii** [comment]
Link to your `setup.py`? I expect you are running `open(x)` instead of `open(x, encoding="utf-8")` in your `setup.py`. You should always specify the encoding when you open any file in text mode - set `PYTHONWARNDEFAULTENCODING` to see the warning. Python 3.15 will fix this by changing the default.

**Comment by lrq3000** [comment]
@henryiii Thank you very much, that was precisely the issue, amazing how you pinpointed it so fast!

Nevertheless, I am a bit confused, because the reason why I did not investigate my `setup.py` is because I now use a `pyproject.toml`, I only keep `setup.py` for 1) cythonization/c extensions compilation, 2) retrocompatibility with Python 2.7.

Hence, I expected `build` to use the definition in `pyproject.toml`, not in `setup.py`! Is this the expected behavior?

You can find my python module which is experiencing this issue here: https://github.com/tomerfiliba-org/reedsolomon/tree/c0d7cb013fe667bb1e2225a53268e873b7f52115

**Comment by henryiii** [comment]
Build reads your pyproject.toml, sets up a virtual env with your `build-system.requires`, then calls the hooks defined in `build-system.build-backend`. Now it's up to setuptools, build isn't in control anymore.

Setuptools runs "setup.py"; that's setuptools' dynamic configuration script. Other backends do not. Anything you set there will override pyproject.toml, since setuptools assumes a dynamic value is more important than a static one.

If all you want to do is support both Python 3 & 2 (I don't recommend supporting Python 2! Supporting Python 2 users hurts Python 3 users & sometimes Python 2 users. If you set python-requires, Python 2 continues to get older versions of your package - I've dropped it a while back in https://github.com/tomerfiliba/plumbum, for example), then I'd use `setup.cfg` for most of the config - that's static, and is supported by older versions of setuptools, so you won't be specifying everything twice. You'll still need setup.py for Cythonizations, but that's it. Also, setuptools has native support for Cythonizations, so you don't need to get stuff from Cython.

**Comment by henryiii** [comment]
Also, you should not handle arguments to `setup.py`, that's very deprecated.

**Comment by lrq3000** [comment]
@henryiii Ha, a fellow "Tomer's project maintainer", nice to meet you :D

I did not think that maintaining compatibility with py2 could actually hurt compatibility for py2 users, you are totally right. I will consider your suggestion, and actually I can say I am pretty certain I will drop support for older versions.

I did not know setuptools ran setup.py after pyproject.toml. I now understand that setup.py takes precedence over other config file sources. Since I will most likely drop py2 support, I will be able to drop all redundant statements in setup.py and only keep the cythonization part.

> Also, you should not handle arguments to setup.py, that's very deprecated.

Don't worry, I am not using setup.py as a CLI anymore :D I am accessing optional build arguments using `--config-setting` which is PEP517 :-)

Thank you very much for your wise suggestions, I will now close this PR as it was due to my misunderstanding of how setuptools works!
